From 1ac0858a4c6a384f039dfc3f8131d86040c14a58 Mon Sep 17 00:00:00 2001
From: Jason Chen <b02280@freescale.com>
Date: Wed, 21 Apr 2010 16:17:31 +0800
Subject: [PATCH] ENGR00122694 v4l2 output: set correct bits_per_pixel for fbi

Uncorrect bits_per_pixel will make wrong fb memaloc size, which will
cause Oops for filling black color operation.

Signed-off-by: Jason Chen <b02280@freescale.com>
---
 drivers/media/video/mxc/output/mxc_v4l2_output.c |   15 +++++++++++++--
 1 files changed, 13 insertions(+), 2 deletions(-)

diff --git a/drivers/media/video/mxc/output/mxc_v4l2_output.c b/drivers/media/video/mxc/output/mxc_v4l2_output.c
index 4c794ab..a8beb77 100644
--- a/drivers/media/video/mxc/output/mxc_v4l2_output.c
+++ b/drivers/media/video/mxc/output/mxc_v4l2_output.c
@@ -1341,8 +1341,19 @@ static int mxc_v4l2out_streamon(vout_data * vout)
 		if (vout->ic_bypass) {
 			pr_debug("Bypassing IC\n");
 			vout->work_irq = -1;
-			fbvar.bits_per_pixel = 8*
-				bytes_per_pixel(vout->v2f.fmt.pix.pixelformat);
+			switch (vout->v2f.fmt.pix.pixelformat) {
+			case V4L2_PIX_FMT_YUV420:
+			case V4L2_PIX_FMT_YVU420:
+			case V4L2_PIX_FMT_NV12:
+				fbvar.bits_per_pixel = 12;
+				break;
+			case V4L2_PIX_FMT_YUV422P:
+				fbvar.bits_per_pixel = 16;
+				break;
+			default:
+				fbvar.bits_per_pixel = 8*
+					bytes_per_pixel(vout->v2f.fmt.pix.pixelformat);
+			}
 			fbvar.nonstd = vout->v2f.fmt.pix.pixelformat;
 		}
 
-- 
1.5.4.4

